src/AppInstallerCLITests/GroupPolicy.cpp

```diff
@@ -266,6 +273,54 @@ TEST_CASE("GroupPolicy_Sources", "[groupPolicy]")
         REQUIRE(policy.value()[0].Data == source.Data);
         REQUIRE(policy.value()[0].Identifier == source.Identifier);
     }
+    SECTION("Source with PinningConfiguration")
+    {
+        using namespace AppInstaller::Certificates;
+
+        auto additionalSourcesKey = RegCreateVolatileSubKey(policiesKey.get(), AdditionalSourcesPolicyKeyName);
+
+        PinningDetails rootCert;
+        rootCert.LoadCertificate(IDX_CERTIFICATE_STORE_ROOT_1);
+        PinningDetails intermediateCert;
+        intermediateCert.LoadCertificate(IDX_CERTIFICATE_STORE_INTERMEDIATE_1);
+        PinningDetails leafCert;
+        leafCert.LoadCertificate(IDX_CERTIFICATE_STORE_LEAF_1);
+
+        auto getBytesString = [](const PinningDetails& details)
+        {
+            std::vector<BYTE> bytes;
+            bytes.assign(details.GetCertificate()->pbCertEncoded, details.GetCertificate()->pbCertEncoded + details.GetCertificate()->cbCertEncoded);
+            return AppInstaller::Utility::ConvertToUTF16(AppInstaller::Utility::ConvertToHexString(bytes));
+        };
+
+        std::wostringstream pinningConfig;
+        pinningConfig << LR"({
+"Chains": [{

```

Comment:
nit: seems bad indentation

Reply:
It is a raw string, but I can bring the beginning down to make it look less awkward.